src/main/java/org/chrisoft/jline4mcdsrv/MinecraftCommandCompleter.java

```diff
@@ -27,14 +27,20 @@ public MinecraftCommandCompleter(CommandDispatcher<ServerCommandSource> cmdDispa
 	@Override
 	public void complete(LineReader reader, ParsedLine line, List<Candidate> candidates)
 	{
-		ParseResults<ServerCommandSource> parseRes = cmdDispatcher.parse(line.line(), cmdSrc);
-		CompletableFuture<Suggestions> cs = cmdDispatcher.getCompletionSuggestions(parseRes, line.cursor());
+		//trim previous and next lines
+		String input = line.line();
+		int left = input.lastIndexOf('\n', line.cursor() - 1); left = left == -1 ? 0 : left + 1;
+		int right = input.indexOf('\n', line.cursor()); right = right == -1 ? input.length() : right;
```

Comment:
Just a random thought:
Sometimes feel it'd be nice if `indexOf` returned an index just after the end of the string (aka `input.length()`) if it couldn't find what it searched for, because then all of this could simply be written as
```
int left = input.lastIndexOf('\n', line.cursor() - 1) + 1;
int right = input.indexOf('\n', line.cursor());
```

The `left` part can actually still be written like this because `left == -1 ? 0 : left + 1` is just `left + 1` but for symmetry's sake I'd leave it like this.

Reply:
I didn't realise ) Just my routine of handling edge cases